binary_and_decimal.rb

```diff
@@ -2,8 +2,18 @@
 #    Fill the array randomly with 0’s and 1’s. Print out the array so that it
 #    appears as a binary number. Calculate the decimal value for this binary
 #    number using the algorithm you devised in class. Print out the decimal value.
+
+
 def binary_to_decimal
-  puts "Not implemented"
+  #creates an array of 8 binary figures
```

Comment:
Minor: The problem statement asks that the method print the binary and decimal values. So, you're just missing the puts statements.
The rest of the logic looks good.